### ISSUE TITLE
transmission-create piece size selection

### ISSUE THREAD

**Issue by Daniel6372** [transmission/transmission#5569]
### What is the issue?

The automatic piece size selection when creating large torrents doesn't go beyond 2 MiB, no matter the size of the torrent. Typically the auto selection should go as high as 16 MiB (higher than this causes issues with some clients).
For example when creating a torrent for files that are about 50 GB total in size, it's much more desirable to have around 3000 x 16 MiB pieces, as opposed to the 24000 x 2 MiB that Transmission automatically selects.
Not sure if this is a bug, but imo it's not ideal behavior.

Auto selection for smaller torrents works as expected with piece sizes 2 MiB and below, as low as 16 KiB being selected when appropriate.

Probably unrelated, manual piece size selection seems to have some issues with torrents of all sizes in the gui:
[test.webm](https://github.com/transmission/transmission/assets/47946277/d3986963-5dba-4deb-802a-11104ce14757)

Manual piece size selection works as expected with cli.

Tested gui on Fedora 38, cli on Fedora 38 and Arch Linux.

### Which application of Transmission?

None

### Which version of Transmission?

4.0.3 (6b0e49bbb2)

**Comment by killemov** [comment]
Is 16 MiB the max? Or can we keep on doubling the size?

**Comment by Daniel6372** [comment]
Piece sizes 32 MiB and 64 MiB are also available, but I think they should not be selected automatically to ensure maximum compatibility with all clients.

**Comment by tearfur** [comment]
@Daniel6372 You may want to create a separate issue for the GTK client.

**Comment by Daniel6372** [comment]
@tearfur you're right, just wanted to confirm it's not specific to GTK**3**, as that's how fedora packages it
https://github.com/transmission/transmission/issues/5617
